src/lib.rs

```diff
@@ -211,25 +211,29 @@ impl Default for IpfsOptions {
     }
 }
 
-#[derive(Debug)]
-pub struct Ipfs<Types: IpfsTypes>(Arc<IpfsInner<Types>>);
-
-impl<Types: IpfsTypes> Clone for Ipfs<Types> {
-    fn clone(&self) -> Self {
-        Ipfs(Arc::clone(&self.0))
-    }
-}
-
 /// Ipfs struct creates a new IPFS node and is the main entry point
 /// for interacting with IPFS.
 #[derive(Debug)]
-pub struct IpfsInner<Types: IpfsTypes> {
+pub struct Ipfs<Types: IpfsTypes> {
     pub span: Span,
```

Comment:
I have missed this earlier, why is this `pub`?

Reply:
the `Span`? good point; I'll check if this can be removed